webapi/src/main/scala/org/knora/webapi/messages/StringFormatter.scala

```diff
@@ -2007,49 +2034,63 @@ class StringFormatter private (
       case None    => None
     }
 
+  def validateAndEscapeOptionalProjectShortname(
+    maybeString: Option[String]
+  ): Validation[ValidationException, Option[String]] =
+    maybeString match {
+      case None        => Validation.succeed(None)
+      case Some(value) => validateAndEscapeProjectShortname(value).map(Some(_))
+    }
+
   /**
    * Given the project shortcode, checks if it is in a valid format, and converts it to upper case.
    *
    * @param shortcode the project's shortcode.
    * @return the shortcode in upper case.
    */
+  @deprecated("Use def validateProjectShortcode(String) instead.")
   def validateProjectShortcode(shortcode: String, errorFun: => Nothing): String = // V2 / value objects
-    ProjectIDRegex.findFirstIn(shortcode.toUpperCase) match {
-      case Some(value) => value
-      case None        => errorFun
-    }
+    validateProjectShortcode(shortcode).getOrElse(errorFun)
+
+  def validateProjectShortcode(shortcode: String): Option[String] =
```

Comment:
None represents an illegal state, so maybe validation would be better here too?

Reply:
In keep it as an `Option`.